AllASPFuncs.py

```diff
@@ -70,107 +70,172 @@
 '''
 
 
-def local_grid(ra_center, dec_center, wcs, size=25, spacing=1.0,
-               image=None, percentiles=[], makecontourGrid=True):
-
+def make_regular_grid(ra_center, dec_center, wcs, size, spacing=1.0,
+                      subsize=9):
     '''
-    Generates a local grid around a RA-Dec center, choosing step size and
-    number of points
+    Generates a regular grid around a (RA, Dec) center, choosing step size.
+
+    ra_center, dec_center: floats, coordinate center of the image
+    wcs: the WCS of the image, currently a galsim.fitswcs.AstropyWCS object
+    spacing: int, spacing of grid points in pixels.
+    subsize: Int, width of the grid in pixels.
+             Specify the width of the grid, which can be smaller than the
+             image. For instance I could have an image that is 11x11 but a grid
+             that is only 9x9.
+             This is useful and different from making a smaller image because
+             when the image rotates, model points near the corners of the image
+             may be rotated out. By taking a smaller grid, we can avoid this.
+
+
+    Returns:
+    ra_grid, dec_grid: 1D numpy arrays of floats, the RA and DEC of the grid.
+
+
     '''
-    Lager.debug('image shape: {}'.format(np.shape(image)))
-    # Build the basic grid
-    subsize = 9  # Taking a smaller square inside the image to fit on
+    if subsize > size:
+        Lager.warning('subsize is larger than the image size. ' +
+                      f'{size} > {subsize}. This would cause model points to' +
+                      ' be placed outside the image. Reducing subsize to' +
+                      ' match the image size.')
```

Comment:
Hmmm.. if you're not going to allow subsize > image size, then conceptually aren't you requiring that all the points are within the circle that inscribes the square to handle all potential rotations?

I.e. saying that subsize == size is fine, makes the first image and its orientation special.  You know that you are creating grid points that will not be constrained by other images at other rotations.